presto-array/src/main/java/com/facebook/presto/array/BlockBigArray.java

```diff
@@ -60,10 +77,13 @@ public void set(long index, Block value)
     {
         Block currentValue = array.get(index);
         if (currentValue != null) {
-            sizeOfBlocks -= currentValue.getRetainedSizeInBytes();
+            // the visitor will change sizeOfBlocks; do not subtract/add the header size in one statement
+            long headerSize = currentValue.getRetainedSizeInBytesBreakdown(decrementVisitor);
```

Comment:
By header I guess you mean object header. To be consistent with the other parts of the code either use instance size or object overhead as I don't think we use object header anywhere else.